From a6718ea2d0812310065f19fda73cb9b76e8b1a36 Mon Sep 17 00:00:00 2001
From: Eric Sun <jian.sun@freescale.com>
Date: Fri, 20 Jul 2012 16:57:10 +0800
Subject: [PATCH 335/505] ENGR00217764 NAND Boot with secure boot doesn't work
 fine

Change signing script so that "0x5a" instead of "0xff" is used as fill
data, thus prevent ROM NAND code mis-understand continous "0xff" in a
page as a critical error and break the U-Boot bootup.

Signed-off-by: Eric Sun <jian.sun@freescale.com>
---
 test/mxc_secureboot/habUimageGen | 4 ++--
 test/mxc_secureboot/habimagegen  | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/test/mxc_secureboot/habUimageGen b/test/mxc_secureboot/habUimageGen
index 06c88f7..d3b60bd 100755
--- a/test/mxc_secureboot/habUimageGen
+++ b/test/mxc_secureboot/habUimageGen
@@ -1,7 +1,7 @@
 #! /bin/bash
 
 echo "extend uImage to 0x3FDFE0..."
-objcopy -I binary -O binary --pad-to 0x3fdfe0 --gap-fill=0xff uImage uImage-pad.bin
+objcopy -I binary -O binary --pad-to 0x3fdfe0 --gap-fill=0x5a uImage uImage-pad.bin
 
 echo "generate IVT"
 ./genIVT
@@ -16,4 +16,4 @@ echo "merge image and csf data..."
 cat uImage-pad-ivt.bin uImage_csf.bin > uImage-signed.bin
 
 echo "extend final image to 0x400000..."
-objcopy -I binary -O binary --pad-to 0x400000 --gap-fill=0xff uImage-signed.bin uImage-signed-pad.bin
+objcopy -I binary -O binary --pad-to 0x400000 --gap-fill=0x5a uImage-signed.bin uImage-signed-pad.bin
diff --git a/test/mxc_secureboot/habimagegen b/test/mxc_secureboot/habimagegen
index af3e8ae..ab92c5c 100755
--- a/test/mxc_secureboot/habimagegen
+++ b/test/mxc_secureboot/habimagegen
@@ -1,7 +1,7 @@
 #! /bin/bash
 
 echo "extend u-boot to 0x2F000..."
-objcopy -I binary -O binary --pad-to 0x2f000 --gap-fill=0xff u-boot.bin u-boot-pad.bin
+objcopy -I binary -O binary --pad-to 0x2f000 --gap-fill=0x5a u-boot.bin u-boot-pad.bin
 
 echo "generate csf data..."
 ../linux/cst --o u-boot_csf.bin < u-boot.csf
@@ -10,6 +10,6 @@ echo "merge image and csf data..."
 cat u-boot-pad.bin u-boot_csf.bin > u-boot-signed.bin
 
 echo "extend final image to 0x31000..."
-objcopy -I binary -O binary --pad-to 0x31000 --gap-fill=0xff u-boot-signed.bin u-boot-signed-pad.bin
+objcopy -I binary -O binary --pad-to 0x31000 --gap-fill=0x5a u-boot-signed.bin u-boot-signed-pad.bin
 
 echo "u-boot-signed-pad.bin is ready"
-- 
1.8.0

